mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -242,6 +244,7 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
         }
 
         EventListenerService.startOrStopService(this)
+        selectServerBasedOnWifi()
```

Comment:
I'm not sure if `onCreate()` is the best location to call `selectServerBasedOnWifi()`. Maybe do it in `onResume()`, but only if the last time the app was open was some time ago?

Reply:
Why not onCreate tho? 🤔

Reply:
The user might close the app by pressing the home button, go to a location with a different known wifi, open the app again and it might be still in memory, so `onCreate()` might not be called.

Reply:
Not sure about the permission state on newer Android phones, but wouldn't it make sense to do it in a BroadcastReceiver?

Reply:
Never mind. Thats not possible anymore. Sorry.

Reply:
> Maybe do it in onResume(), but only if the last time the app was open was some time ago?

I'd do it in onStart, and compare current Wifi SSID to the one from the last check ... so base it on actual SSID change, not time.

Reply:
Should the last check be saved in memory or shared prefs?

Reply:
IMHO memory, since you *do* want this to happen when opening the app when it wasn't running before, regardless of the state before app shutdown.